client/state/data-layer/wpcom/sites/hosting/php-version.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+import {
+	HOSTING_PHP_VERSION_GET,
+	HOSTING_PHP_VERSION_GET_SUCCESS,
+	HOSTING_PHP_VERSION_SET,
+	HOSTING_PHP_VERSION_SET_FAILURE,
+	HOSTING_PHP_VERSION_SET_SUCCESS,
+} from 'state/action-types';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { translate } from 'i18n-calypso';
+import { composeAnalytics, recordGoogleEvent, recordTracksEvent } from 'state/analytics/actions';
+
+const updateNoticeId = 'hosting-php-version';
+
+const getPhpVersion = action =>
+	http(
+		{
+			method: 'GET',
+			path: `/sites/${ action.siteId }/hosting/php-version`,
+			apiNamespace: 'wpcom/v2',
+			body: {},
+		},
+		action
+	);
+
+const getPhpVersionSuccess = ( action, version ) => {
+	return {
+		type: HOSTING_PHP_VERSION_GET_SUCCESS,
+		siteId: action.siteId,
+		version,
+	};
+};
+
+const updatePhpVersion = action =>
+	http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/hosting/php-version`,
+			apiNamespace: 'wpcom/v2',
+			body: {
+				version: action.version,
+			},
+		},
+		action
+	);
+
+export const hostingPhpVersionUpdateTracking = ( version, result ) =>
```

Comment:
Analytics can be set as meta on any action, if there is a meaningful one to attach to. Should be fine in this case, but we don't always want to  append analytics in the data-layer, if we're interested in calling out different usages in component UI

So this might look like:

```jsx
const updatePhpVersion = () => {
	withAnalytics(
		composeAnalytics(
		recordGoogleEvent(
			'Hosting Configuration',
			'Clicked "Update PHP Version" Button in PHP Version box',
			`PHP Version Update ${ result }`,
			version
		),
		recordTracksEvent( 'calypso_hosting_configuration_php_version_update', {
			result,
			version,
		} )
	);
		updateAtomicPhpVersion( siteId, currentUserId )
	);

Reply:
Problem here is `result`. :) If we want to track the result, won't work w/o `useEffect` or multiple actions.

Reply:
I meant, if we wanted this in the component. :)